src/Adapter/Manufacturer/CommandHandler/AddManufacturerHandler.php

```diff
@@ -0,0 +1,149 @@
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Manufacturer\CommandHandler;
+
+use Exception;
+use Manufacturer;
+use PrestaShop\PrestaShop\Adapter\Language\LanguageDataProvider;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Command\AddManufacturerCommand;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Command\Handler\AddManufacturerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Manufacturer\Exception\CannotAddManufacturerException;
+
+/**
+ * Adds a new manufacturer
+ */
+class AddManufacturerHandler implements AddManufacturerInterface
+{
+    /**
+     * @var AddManufacturerCommand
+     */
+    private $command;
+
+    /**
+     * @var LanguageDataProvider
+     */
+    private $languagesDataProvider;
+
+    /**
+     * Index of installed languages as $langId => true
+     * @var array
+     */
+    private $installedLanguages = [];
+
+    /**
+     * @param AddManufacturerCommand $command
+     *
+     * @throws CannotAddManufacturerException
+     *
+     * @return int Id of the added Manufacturer
+     */
+    public function handle(AddManufacturerCommand $command)
+    {
+        $this->command = $command;
+        try {
+            $this->loadInstalledLanguages();
+
+            return $this->buildManufacturer();
+        } catch (Exception $e) {
+            throw new CannotAddManufacturerException(
+                "Unable to add a new Manufacturer",
+                0,
+                $e
+            );
+        }
+    }
+
+    /**
+     * @return int Id of the new Manufacturer
+     *
+     * @throws Exception
+     */
+    private function buildManufacturer() {
+        $command = $this->command;
+        $entity = new Manufacturer();
+
+        $entity->name = $command->getName();
+
+        $this->assignMultilangProperties($entity);
+
+        $entity->validateFields();
+
+        $entity->add();
+
+        if ($entity->id <= 0) {
+            throw new Exception(
+                sprintf('Invalid entity id after creation: %s', var_export($entity->id, true))
```

Comment:
prefer `print_r` which is a function for debug (human-readable information) when `var_export` cost more and isn't for debug :)

Reply:
The advantage of `var_export` is that it adds quotes for strings (very useful for empty strings or spaces), and displays null as NULL (where `print_r` displays an empty string).

https://3v4l.org/DeiRc

```
string using print_r:    this is a string 
string using var_export: 'this is a string'

int using print_r:    123 
int using var_export: 123

array using print_r:    Array
(
    [0] => foo
    [bar] => baz
)
 
array using var_export: array (
  0 => 'foo',
  'bar' => 'baz',
)

object using print_r:    stdClass Object
(
    [0] => foo
    [bar] => baz
)
 
object using var_export: stdClass::__set_state(array(
   0 => 'foo',
   'bar' => 'baz',
))

null using print_r:     
null using var_export: NULL

empty string using print_r:     
empty string using var_export: ''

space using print_r:     
space using var_export: ''
```